theano/gpuarray/opt.py

```diff
@@ -2069,7 +2085,7 @@ def provide_inputs(self, node, inputs):
                                          node.op.subsample,
                                          node.op.filter_dilation)
 
-            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[2:])
+            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[-2:])
```

Comment:
Also the meta-optimiser was written in such a way that if one optimiser causes an error, it will select another one